pkg/internal/ingress/v1beta1.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+	"fmt"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	listersv1beta1 "k8s.io/client-go/listers/networking/v1beta1"
+)
+
+const ConvertedGVKAnnotation = `internal.cert-manager.io/converted-gvk`
```

Comment:
As the type used throughout cert-manager is now `networking.k8s.io/v1` `Ingress`, the ingress-shim sync function needs to know that the ingress it's looking at was converted from a `v1beta1` to a `v1` to set the owner reference on the resulting certificate correctly. This will only be used on k8s 1.16.

Reply:
By Kubernetes 1.16, do you mean 1.16, 1.17, and 1.18 since v1 was introduced with Kubernetes 1.19?

Would it be possible to add it as a comment? 

```suggestion
// On Kubernetes 1.16, 1.17, and 1.18, the `networking.k8s.io/v1` did not exist yet.
// For these Kubernetes versions, cert-manager converts `networking.k8s.io/v1beta1`
// into `networking.k8s.io/v1`, which is the one version used across cert-manager.
// 
// But this conversion means that we lose the "original" apiVersion, which is needed
// by the ingress-shim sync function to set the correct `ownerReference` on the
// Certificate it creates.
//
// This annotation is only set on Kubernetes 1.16, 1.17, and 1.18.
const ConvertedGVKAnnotation = `internal.cert-manager.io/converted-gvk`
```